### PR TITLE
OC10878:Participate access code viewed" is getting recorded in audit …

### PR DESCRIPTION
**Pull request by Taoli2018**
…log just by clicking on view accessCode link

### CODE REVIEW

File: web/src/main/java/org/akaza/openclinica/service/UserServiceImpl.java [hunk 1]
```diff
@@ -599,12 +599,17 @@ public ParticipantAccessDTO getAccessInfo(String accessToken, String studyOid, S
             logger.error("Participant account not found");
             return null;
         }
-        String accessCode = keycloakClient.getAccessCode(accessToken, pUserAccount.getUserUuid(), customerUuid);
-
-        if (accessCode == null) {
-            logger.error(" Access code from Keycloack returned null ");
-            return null;
+
```
Comment (svadla-oc) [inline web/src/main/java/org/akaza/openclinica/service/UserServiceImpl.java]: Create a new method "public ParticipantAccessDTO getAccessInfo(String accessToken, String studyOid, String ssid, String customerUuid, UserAccountBean userAccountBean,boolean auditAccessCodeViewing, boolean includeAccessCode)" and use that to control whether to call keycloak to get access code.
Update the original method to call the new method and set "includeAccessCode" to true.

File: web/src/main/java/org/akaza/openclinica/controller/UserController.java [hunk 2]
```diff
@@ -165,13 +165,20 @@ public ModelMap userHandler(HttpServletRequest request,
     }
 
     @RequestMapping( value = "/clinicaldata/studies/{studyOID}/participants/{SSID}/accessLink", method = RequestMethod.GET )
-    public ResponseEntity<ParticipantAccessDTO> getAccessLink(HttpServletRequest request, @PathVariable( "studyOID" ) String studyOid, @PathVariable( "SSID" ) String ssid) {
+    public ResponseEntity<ParticipantAccessDTO> getAccessLink(HttpServletRequest request, @PathVariable( "studyOID" ) String studyOid, @PathVariable( "SSID" ) String ssid,
+    		@RequestParam( value = "accessCode", defaultValue = "n", required = false ) String accessCode) {
```
Comment (svadla-oc) [inline web/src/main/java/org/akaza/openclinica/controller/UserController.java]: Rename this to "includeAccessCode"

File: web/src/main/webapp/WEB-INF/jsp/managestudy/viewStudySubject.jsp [hunk 3]
```diff
@@ -1742,10 +1742,10 @@
             $('#connect-button').attr('disabled', 'disabled');
     }
 
-    function getAccessCode() {
+    function getAccessCode(accessCode) {
```
Comment (svadla-oc) [inline web/src/main/webapp/WEB-INF/jsp/managestudy/viewStudySubject.jsp]: Rename this to includeAccessCode

**Comment by jmcinerney** [comment]
jenkins retest this please

**Comment by svadla-oc** [comment]
👍

**Pull request merged**